iothub/device/tests/Authentication/EdgeDeviceOnBehalfOfTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Text;
+using System.Threading.Tasks;
+using System;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client.Authentication;
+
+namespace Microsoft.Azure.Devices.Client.Tests.OnBehalfOf
+{
+    public class EdgeDeviceOnBehalfOfTests
+    {
+        private static string _testKey => Convert.ToBase64String(Encoding.UTF8.GetBytes(Guid.Empty.ToString("N")));
+
+        [TestMethod]
+        public async Task ConnectDeviceOnBehalfOf()
+        {
+            string leafDeviceId = "test-leaf-device";
+
+            DeviceConnectionSettings edgeDevice = new DeviceConnectionSettings("e4k-hub.azure-devices.net", "test-edge-device")
+            {
+                SharedAccessKey = _testKey
+            };
+
+            var edgeHubCs = await EdgeHubConnectionSettingsBuilder.Build(edgeDevice);
+
+            IAuthenticationMethod leafAuth = new ClientAuthenticationForEdgeHubOnBehalfOf(
+                edgeHubCs.SharedAccessKey!,
+                edgeHubCs.DeviceId!,
+                leafDeviceId,
+                null,
+                TimeSpan.FromMinutes(10),
+                5);
+
+            IotHubDeviceClient leafClient = new(edgeHubCs.HostName, leafAuth,
+                new IotHubClientOptions(
+                    new IotHubClientAmqpSettings
```

Comment:
should we test with MQTT as well?

we dont need to configure multiplexing in any test